src/EngineManager.php

```diff
@@ -59,6 +65,31 @@ protected function ensureAlgoliaClientIsInstalled()
         throw new Exception('Please install the Algolia client: algolia/algoliasearch-client-php.');
     }
 
+    /**
+     * Set the default Algolia config headers.
+     *
+     * @return array
+     */
+    protected function defaultAlgoliaHeaders()
+    {
+        if (! config('scout.user')) {
+            return [];
+        }
+
+        $headers = [];
+        $request = $this->container->make(Request::class);
+
+        if (filter_var($ip = $request->ip(), FILTER_VALIDATE_IP, FILTER_FLAG_NO_PRIV_RANGE | FILTER_FLAG_NO_RES_RANGE)) {
```

Comment:
If there is no request at all ( testing, cli ), do we still want to modify the headers with his information?

Reply:
It should already strip things out for ips like 127.0.0.1 etc because of the `filter_var` so I think we're good here.

Reply:
I've added a `! config('app.debug')` check.